### DISCUSSION TITLE
problems with mybinder requirements.txt

### DISCUSSION THREAD

**Discussion by thomasfermi** [executablebooks/meta#190] [Q&A] [answered]
Hi everyone,

I somehow can't get interactivity via Thebe/Binder to work. I followed the ínstructions in ["Launch buttons for interactivity"](https://jupyterbook.org/interactive/launchbuttons.html). It did not say where to put the requirements.txt, so I checked how it was done in [https://github.com/executablebooks/jupyter-book](https://github.com/executablebooks/jupyter-book). I mimicked that for my [repo](https://github.com/thomasfermi/Algorithms-for-Automated-Driving), but when I click the Binder link in my jupyter book on [this section](https://thomasfermi.github.io/Algorithms-for-Automated-Driving/Control/PID.html) it takes me to a "non-working" Binder: When I execute the notebook it says that the numpy import failed.

Can someone tell me what I am doing wrong? Is it something obvious? If not, is something wrong with my [ipynb file](https://github.com/thomasfermi/Algorithms-for-Automated-Driving/blob/master/book/Control/PID.ipynb)? Maybe the metadata or something?

Regarding the requirements.txt: I do my actual requirements managing with conda, that is why there is an environment.yml file in the "book" folder (for github actions workflow). For Binder I do not need all the stuff from my conda environment: pytorch is in my environment.yml, but the section where I want "Thebe"-interactivity on my web page only needs numpy+matplotlib. That is why I just copied the content of the [requirements.txt from the docs folder of the jupyter-book repo ](https://github.com/executablebooks/jupyter-book/blob/master/docs/requirements.txt).

I would be really grateful for some help :)

P.S.: I really love jupyter book. Keep up the great work :)!

**Comment by mmcky** [comment, marked as answer]
hi @thomasfermi -- do you know if your repository builds directly with mybinder.org? The jupyter-book `binder` setup is contained in `.binder` at the root level of the repository. It looks like you have put `requirements.txt` in `books` and I don' think binder would find it in that location. 

If you haven't tried this -- I would start there to see if the project is building by adding in the link to your repository, branch etc. and then click on launch. https://mybinder.readthedocs.io/en/latest/introduction.html#build-your-repository

Please let me know how you go.

**Reply by mmcky** [reply]
cool repo btw :-)

**Reply by thomasfermi** [reply]
Hi @mmcky and thanks a lot for your help! Putting the `requirements.txt` into a `.binder` directory at root level fixed my problem. I even found that it works with an `environment.yml` instead of a `requirements.txt`. I ran into a new problem however:

* In this [section of my book](https://thomasfermi.github.io/Algorithms-for-Automated-Driving/Control/PID.html) I can launch "Binder" and it works: I get plots and I can manipulate the plots with `ipywidgets` sliders
* If I launch "Live Code" instead of "Binder" it doesn't work as expected: Plots get generated but the `ipywidgets` sliders disappear as soon as I click "run" in the corresponding cell.

Am I missing something, or should I file this as a bug report for Thebe?

P.S.: I am glad that you like the repo! Thanks :-)

**Reply by thomasfermi** [reply]
Update: I've been browsing github and it seems that `interact` does not work in Thebe, and it is a [known issue](https://github.com/executablebooks/thebe/issues/111). I will have to wait until this gets fixed.

Thanks again for the help @mmcky  :-)

**Reply by mmcky** [reply]
thanks for the update @thomasfermi -- I am glad you're up and running (half-way at least). Thanks for linking that `thebe` issue here.
